LGV Forecasting Inputs Growth

wsp-mbuckley opened the pull request:
- Added script to grow LGV model inputs.
- Fixed issues with LGV commute segment not being grown correctly and double counting

wsp-mbuckley committed:
Added optional encoding parameter to VOA read function

Updated `voa_ratings_list` to pass encoding parameter to `read_csv`.

wsp-mbuckley committed:
Merge branch 'master' into voa_data_replacement

wsp-mbuckley committed:
Updated to pandas v1.4

wsp-mbuckley committed:
Updated LGV config to use caf.toolkit

Added `caf.toolkit` dependancy.
Updated `LGVInputPaths` to be a child of `BaseConfig` so it can read /
write and validate YAML config files.
Removed unneeded `LGVConfig` class.
Updated writing example config.

wsp-mbuckley committed:
Added pyproject TOML file

Updated some of pylint settings and black settings.

wsp-mbuckley committed:
Replaced VOA with warehouse floorspace for delivery

Removed function for loading VOA data.
Added function for loading LSOA warehouse data.
Removed unneeded metadata for VOA input.
Renamed input parameters from VOA to warehouse.

wsp-mbuckley committed:
LGV - replace VOA data for commute segment

Added 3 new warehouse inputs for commute segment to replace VOA dataset.
Updated commute segment to get 3 CSVs of warehouse floorspace and apply
different weighting factors to each when calculating trips.
Tidied up config file for warehouse and bres paths.

wsp-mbuckley committed:
Merge remote-tracking branch 'origin/master' into voa_data_replacement

wsp-mbuckley committed:
Updated LGV model UI for new warehouse data

Added the commute warehouse high, medium and low inputs to UI.
Removed unneeded `LGVInputsUI` class and instead had the UI create
`LGVInputPaths` instance itself.
Write LGV config values to output folder.

wsp-mbuckley committed:
LGV delivery depot infilling

Added depot infilling back to delivery segment.

wsp-mbuckley committed:
LGV commute warehouse infilling

Updated commute segment to infill warehouse data for given zones.
Fixed delivery segment warehouse infilling.
Added check to make sure trip end outputs include all model zones.

wsp-mbuckley committed:
LGV fixed depreciation warnings

wsp-mbuckley committed:
Added op diagrams

Added current Draw.io op diagrams, LGV one to be updated.

wsp-mbuckley committed:
LGV updated README new warehouse data

Updated README and flowcharts for the new warehouse input data to
replace VOA.

wsp-mbuckley committed:
Removed conda defaults channel

wsp-mbuckley committed:
Adding LGV forecast inputs script

WIP converting Jupyter notebook into Python script, WIP.

wsp-mbuckley committed:
LGV model type hint and log message

wsp-mbuckley committed:
LGV forecast inputs growth script

Finished writing script to grow unavailable LGV model inputs to
forecast year.
Added functions for reading the commute segment inputs

wsp-mbuckley committed:
LGV forecast inputs - updated grown input paths summary

wsp-mbuckley committed:
Run LGV forecast inputs batch file

Moved code to activate conda environment to new batch file and created
batch file to run LGV forecast inputs script.

wsp-mbuckley committed:
LGV model - fixed loading households

Filter "Authority" and "County" from household projections inputs which
come from TEMPro.

wsp-mbuckley committed:
LGV forecast inputs - fixed Scotland occupation growth

Fixed error when calculating average growth to apply to Scotland.

wsp-mbuckley committed:
LGV model - fixed QS606 header number

wsp-mbuckley committed:
Added defaults channel to environment.yml

Removing defaults channel from environment seemed to cause issues with
conda so reverting it to fix the issues for now.

wsp-mbuckley committed:
LGV forecast inputs - fixed dwellings and occupation

Added renaming columns of grown data to forecast year.

wsp-mbuckley committed:
LGV - fixed commute segment inputs

Fixed reading NDR and TEMPro household projections.

wsp-mbuckley committed:
LGV forecast inputs - output summary

Created new summary spreadsheet showing differences between base and
forecasted inputs.

wsp-mbuckley committed:
LGV model - fixed commute and delivery growth

Applied LGV growth factor to deilvery trips.
Fixed bug which didn't apply LGV growth to commute land use trips.

wsp-mbuckley committed:
LGV model - delivery growth factor

Added specific growth factor for delivery trips because the inputs are
the base year not the survey year.

wsp-mbuckley committed:
LGV model - fixed commute growth

Fixed commute growth factor bug.
Updated LGV forecast inputs to output base to forecast growth factors.

wsp-mbuckley committed:
LGV forecast inputs - output folder

wsp-mbuckley committed:
LGV forecast inputs - fleet projections input

Added growth calculations using NoCARB fleet projections.

SimonTfN committed:
Added trip ends automation script

Sums totals for trip ends and puts them in a singular csv

wsp-mbuckley committed:
Fixed LGV parameters depot infill

Fixed depot infill parameters for non-integer zone systems

wsp-mbuckley committed:
LGV - Fixed gravity model with non-int zones

Fixed GM inputs expecting zones to be integers.

wsp-mbuckley committed:
Merge branch 'lgv_forecasting_inputs_script' of https://github.com/Transport-for-the-North/caf-freight-tools into lgv_forecasting_inputs_script

wsp-mbuckley committed:
LFT - sphinx docs update

Added README to sphinx docs and updated template.

wsp-mbuckley committed:
LGV personal segment placeholders and todo

Added placeholder functions for the personal segment update and todo comments for tasks to be completed.

SimonTfN committed:
Adding personal segments

Added personal matrix production into the model. Adjusted config file and inputs accordingly.

SimonTfN committed:
Apply suggestions from code review

Co-authored-by: wsp-mbuckley <72382942+wsp-mbuckley@users.noreply.github.com>

SimonTfN committed:
Small doc string changes

SimonTfN committed:
very minor description change

wsp-mbuckley committed:
Merge pull request #19 from Transport-for-the-North/lgv_personal_segment

Adding LGV personal segment

wsp-mbuckley committed:
LGV personal GUI inputs

Added inputs for LGV personal to GUI.
Updated GUI to use 3 columns of inputs.
Added ListInput widget.

SimonTfN committed:
Merge pull request #20 from Transport-for-the-North/lgv_personal_gui

LGV personal GUI inputs

SimonTfN committed:
Updated Scotland Wales dwellings input

Scotland and Wales dwellings data was taking into account future year instead of previous year. This has been adjusted.

SimonTfN committed:
Updating Scotland-Wales dwelling

Noticed small error, has been corrected.

MattBuckley-TfN committed:
Added dummy personal matrix if it fails

Wrapped producing personal matrix in try / except which will produce a dummy matrix of ones if any exceptions occur.

Caitlin-Stokes reviewed (approved):
A couple of backlog items but otherwise no issues with the code!

MattBuckley-TfN committed:
Merge remote-tracking branch 'origin/master' into lgv_forecasting_inputs_script

MattBuckley-TfN committed:
Merge branch 'master' into lgv_forecasting_inputs_script

MattBuckley-TfN committed:
Updated activate environment batch

Calling the conda activate script to setup conda before activating environment.
